leaf-ui/rappid-extensions/ElementInspector.js

```diff
@@ -219,38 +225,48 @@ var ElementInspector = Backbone.View.extend({
-            // Stochastic should be the only option for '(no value)'
-            this.$('.function-type').empty();
-            this.$('.function-type').append('<option value=R> Stochastic </option>');
-        } else {
-            this.$('.function-type').empty();
-            this.$('.function-type').append('<option value=NT> No Function </option>');
-            this.$('.function-type').append('<option value=C> Constant </option>');
-            this.$('.function-type').append('<option value=R> Stochastic </option>');
-            this.$('.function-type').append('<option value=I> Increase </option>');
-            this.$('.function-type').append('<option value=D> Decrease </option>');
-            this.$('.function-type').append('<option value=RC> Stochastic-Constant </option>');
-            this.$('.function-type').append('<option value=CR> Constant-Stochastic </option>');
-            this.$('.function-type').append('<option value=MP> Montonic Positive </option>');
-            this.$('.function-type').append('<option value=MN> Montonic Negative </option>');
-            this.$('.function-type').append('<option value=SD> Satisfied Denied </option>');
-            this.$('.function-type').append('<option value=DS> Denied Satisfied </option>');
-            this.$('.function-type').append('<option value=UD> User Defined </option>');
+    */
+  checkInitialSatValue: function() {
+
+          if (this.model.get('intention').getInitialSatValue() == '(no value)') {
+          // Stochastic should be the only option for '(no value)'
+              this.$('.function-type').empty();
+              this.$('.function-type').append('<option value=R> Stochastic </option>');
+          } else {
+             this.$('.function-type').empty();
+             this.$('.function-type').append('<option value=NT> No Function </option>');
+             this.$('.function-type').append('<option value=C> Constant </option>');
+             this.$('.function-type').append('<option value=R> Stochastic </option>');
+             this.$('.function-type').append('<option value=I> Increase </option>');
+             this.$('.function-type').append('<option value=D> Decrease </option>');
+             this.$('.function-type').append('<option value=RC> Stochastic-Constant </option>');
+             this.$('.function-type').append('<option value=CR> Constant-Stochastic </option>');
+             this.$('.function-type').append('<option value=MP> Montonic Positive </option>');
+             this.$('.function-type').append('<option value=MN> Montonic Negative </option>');
+             this.$('.function-type').append('<option value=SD> Satisfied Denied </option>');
+             this.$('.function-type').append('<option value=DS> Denied Satisfied </option>');
+             this.$('.function-type').append('<option value=UD> User Defined </option>');   
         }
+
     },
 
     nameAction: function(event) {
+
```

Comment:
remove line. also this predates you but ENTER_KEY is only used once so maybe just remove the variable and replace with 13 directly in the below if statement and then just comment what it means

Reply:
Do you know what it means/ what it's used for?

Reply:
Yeah, 13 is just the code that corresponds to the enter key, and its just so that when you press enter, the code knows to update the name (I believe)